.rubocop.yml

```diff
@@ -1,6 +1,13 @@
 # Enable this when the VSC rubocop extension supports rails
 #inherit_from: ./config/rails.rubocop.yml
 
+AllCops:
+  # Exclude automatically generated rails files for easier app:update's etc
+  Exclude:
+    - bin/*
+    - db/**/*
+    - config/**/*
```

Comment:
I'm not convinced that we should be excluding migrations and configurations (though definitely have to exclude schema). Migrations are all hand written at the end of the day, and similarly configurations should be consistent.

Reply:
I do agree that these directives are quite aggressive. And will change to narrow them down to specific files instead. See the following for my motivations though:

Regarding `config/`:
I guess some initializers might be hand-written. But even those tend to be generated by gems via `rake my_gem:install`. The motivation for ignoring comes from experience of maintaining loads of rails apps across more than a decade. Eventually you end up wanting your `rails app:update` and similar commands for gems to provide clean and easily managed diffs rather than adhere to in house styling.

Regarding migrations:
I always start off with `rails generate migration` or `rails generate model` etc to get going with migrations. If you follow conventions using the command line you can get the entire file including the attributes you want to create / modify generated on the fly. So I rarely do anything manually in these files, apart from adding indexes.

I do believe rubocops auto-formatting can alleviate some of this pain these days if consistently applied after generating files... So I guess that's an alternative to consider.

Reply:
Are you sure? Rubocop docs seem to say otherwise. Groups are separated by an empty newline (or we could set it to be separated by comments if preferred).
https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Bundler/OrderedGems

Reply:
I actually tried to fix the error by sorting the gems within the commented boundaries but still got the hound error... maybe I was tripping out or something.

I might create a separate PR to sort all the gems to keep this branch a bit more focused and verify the rubocop behaviour.